### DISCUSSION TITLE
In the @supabase/ssr auth walkthrough, the createServerClient helper function takes cookieStore as an argument—is this a requirement, or can we call cookies() from within the helper function?

### DISCUSSION THREAD

**Discussion by xvvvyz** [supabase/supabase#21299] [Questions] [answered]
Read: https://supabase.com/docs/guides/auth/server-side/nextjs

It would make calling the helper slightly more convenient.

**Comment by charislam** [comment, marked as answer]
Great question. You can!

When we wrote this, there was some uncertainty about how Next.js determines that a route is dynamic, so we thought it was safest to call `cookies()` directly from the Server Component. We did a bit more digging, and we now think it'll work perfectly fine (as long as you don't [move the call out of context](https://nextjs.org/docs/messages/dynamic-server-error)), so we'll be updating the guide soon to do exactly as you suggest! 😄
